tracing/src/macros.rs

```diff
@@ -783,6 +783,123 @@ macro_rules! event {
     );
 }
 
+/// Checks whether a span or event is enabled depending on the provided metadata.
+///
+/// This macro is a specialized tool: it is intended to be used prior
+/// to an expensive computation required *just* for that event, but
+/// *cannot* be done as part of an argument to that event, such as
+/// when multiple events are emitted (e.g., iterating over a collection
+/// and emitting an event for each item).
+///
+/// ## Usage
```

Comment:
nit: i think generally these should be level-1 headings:
```suggestion
/// # Usage
```